File: 04 - ORIENTAÇÃO A OBJETOS/Java/Exercicio 1/PrincipalCaixaDeNumero.java
/*
1 – Elabore um programa “Caixa de Numero” contendo duas classes no qual deve
gerar um valor randômico de 1 a 10, dizer se o numero é par, mostrar a
tabuada desse número e exibir esse número secreto no final.
 */

import java.util.Random;

public class PrincipalCaixaDeNumero {
    public static void main(String[] args) {
        CaixaDeNumero caixa = new CaixaDeNumero();
        Random random = new Random();
        
        caixa.numeroSecreto = random.nextInt(10);
        if(caixa.ehPar())
            System.out.println("O número é par!");
        else
            System.out.println("Não é par!");
        
        System.out.println("Tabuada: ");
        int[] vetResp = caixa.dizerTabuada();
        for(int i : vetResp)
            System.out.println(i);
        
        System.out.println("\n"+caixa.dizerQualNumero());
    }
}

File: 04-ORIENTACAO_A_OBJETOS/JavaScript - NodeJS/Exercicio 2/Com Pacotes/PrincipalCaixaDeNumero.js
let CaixaDeNumero = require('./CaixaDeNumero')

function teste(incio, fim, palpite){
    var numeroSecreto = Math.floor(Math.random() * 10)
    let caixa = new CaixaDeNumero(numeroSecreto)
    var op=1
    do{
        switch(op){
            case 1:
                console.log(caixa.ehPar())
                break
            case 2:
                caixa.estaEntre(incio, fim)
                break
            case 3:
                caixa.arriscaPalpite(palpite)
                break
            case 4:
                console.log('A tabuada: '+caixa.dizerTabuada())
                break
            case 5:
                console.log('O número secreto é: '+caixa.dizerNumeroSecreto())
                break
            case 6:
                console.log('Saindo...\n')
                break
            default:
                console.log('Opção incorreta!')
                break
        }
        op++
    } while(op != 7)
}

teste(2, 6, 4)
teste(1, 5, 8)

